daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionCoderSpec.scala

```diff
@@ -172,10 +179,26 @@ class TransactionCoderSpec
               .decodeVersionedTransaction(defaultNidDecode, defaultCidDecode, _))) {
               case (Right(decWithMin), Right(decWithMax)) =>
                 decWithMin.transaction shouldBe (if (txvMin precedes minExerciseResult)
-                                                   withoutExerciseResult(tx)
+                                                   transactionWithout(tx, { x: Tx.Node =>
+                                                     withoutExerciseResult(
+                                                       withoutContractKeyInExercise(x))
+                                                   })
+                                                 else if (txvMin precedes minContractKeyInExercise)
+                                                   transactionWithout(tx, { x: Tx.Node =>
+                                                     withoutContractKeyInExercise(x)
+                                                   })
                                                  else tx)
                 decWithMin.transaction shouldBe (if (txvMin precedes minExerciseResult)
-                                                   withoutExerciseResult(decWithMax.transaction)
+                                                   transactionWithout(decWithMax.transaction, {
+                                                     x: Tx.Node =>
+                                                       withoutExerciseResult(
+                                                         withoutContractKeyInExercise(x))
+                                                   })
+                                                 else if (txvMin precedes minContractKeyInExercise)
+                                                   transactionWithout(decWithMax.transaction, {
+                                                     x: Tx.Node =>
+                                                       withoutContractKeyInExercise(x)
+                                                   })
```

Comment:
You're making me feel bad about introducing this pattern in the first place :)

Would you  mind factoring this pattern out even further? Taking the transaction and the txVer/txvMin.

Reply:
Something like

```scala
  def minimalistTx[Nid, Cid, Val](txvMin: TransactionVersion, tx: GenTransaction[Nid, Cid, Val]): GenTransaction[Nid, Cid, Val] =
        if (txvMin precedes minExerciseResult)
                                                   transactionWithout(tx, { x: Tx.Node =>
                                                     withoutExerciseResult(
                                                       withoutContractKeyInExercise(x))
                                                   })
                                                 else if (txvMin precedes minContractKeyInExercise)
                                                   transactionWithout(tx, { x: Tx.Node =>
                                                     withoutContractKeyInExercise(x)
                                                   })
                                                 else tx
```

Reply:
thanks, done.